### PR TITLE
Fixed NullPointerException when formController.getInstancePath() is null

### LINKED ISSUE #1197: Ex String Widget, Ex Integer Widget, Ex Decimal Widget with `readonly=true()` option allow to use paste option
#### Software and hardware versions 
Collect v1.8.0 beta 10, 
Android v4.1, v4.2, v4.4, v5.1, v6.0

#### Problem description
Ex String Widget, Ex Integer Widget, Ex Decimal Widget with `readonly=true()` option allow to use paste option
#### Steps to reproduce the problem
1. User starts job with Ex String Widget, Ex Integer Widget, Ex Decimal Widget
2. Questions are read only
3. User tap and hold on question
4. User is able to paste some value

#### Expected behavior
Paste option is blocked for read only questions

#### Other information 
Discovered during testing [StringWidget/IntegerWidget/DecimalWidget with option allow to use paste option](https://github.com/opendatakit/collect/pull/1193)

### LINKED ISSUE #1997: Exception java.lang.NullPointerException:
> org.odk.collect.android.activities.FormEntryActivity.isInstanceComplete (FormEntryActivity.java:2739)
org.odk.collect.android.activities.FormEntryActivity.onOptionsItemSelected (FormEntryActivity.java:996)
android.app.Activity.onMenuItemSelected (Activity.java:2652)
android.support.v4.app.FragmentActivity.onMenuItemSelected (FragmentActivity.java:380)
android.support.v7.app.AppCompatActivity.onMenuItemSelected (AppCompatActivity.java:195)
android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected (WindowCallbackWrapper.java:108)
android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected (WindowCallbackWrapper.java:108)
android.support.v7.app.ToolbarActionBar$2.onMenuItemClick (ToolbarActionBar.java:63)
android.support.v7.widget.Toolbar$1.onMenuItemClick (Toolbar.java:203)
android.support.v7.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected (ActionMenuView.java:780)
android.support.v7.view.menu.MenuBuilder.dispatchMenuItemSelected (MenuBuilder.java:822)
android.support.v7.view.menu.MenuItemImpl.invoke (MenuItemImpl.java:171)
android.support.v7.view.menu.MenuBuilder.performItemAction (MenuBuilder.java:973)
android.support.v7.view.menu.MenuBuilder.performItemAction (MenuBuilder.java:963)
android.support.v7.widget.ActionMenuView.invokeItem (ActionMenuView.java:624)
android.support.v7.view.menu.ActionMenuItemView.onClick (ActionMenuItemView.java:150)
android.view.View.performClick (View.java:4457)
android.view.View$PerformClick.run (View.java:18491)
android.os.Handler.handleCallback (Handler.java:733)
android.os.Handler.dispatchMessage (Handler.java:95)
android.os.Looper.loop (Looper.java:136)
android.app.ActivityThread.main (ActivityThread.java:5336)
java.lang.reflect.Method.invokeNative (Method.java)
java.lang.reflect.Method.invoke (Method.java:515)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:871)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:687)
dalvik.system.NativeStart.main (NativeStart.java)

https://console.firebase.google.com/u/0/project/api-project-322300403941/monitoring/app/android:org.odk.collect.android/cluster/e8e3a0da?reportTypes=FATAL_JAVA&duration=2592000000&appVersions=2705

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1997 

#### What has been done to verify that this works as intended?
I'm not able to reproduce the issue so I just added a null check.

#### Why is this the best possible solution? Were any other approaches considered?
It's just a null check.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/dao/helpers/InstancesDaoHelper.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public static boolean isInstanceComplete(boolean end) {
         boolean complete = false;
 
         FormController formController = Collect.getInstance().getFormController();
-        if (formController != null) {
+        if (formController != null && formController.getInstancePath() != null) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/dao/helpers/InstancesDaoHelper.java]: Would be good to update the log message on line 65 as well to make it easier to track down any possible problems.

**Comment by grzesiek2010** [comment]
@lognaturel you are right I fixed it and I also added naming improvements replacing `instancePath` filed name with `instanceFile` since it's for Files, not Strings.

**Comment by grzesiek2010** [comment]
I fixed the problem visible in reports but there are a few other places where we don't check for null before to avoid crashing so maybe this is a good opportunity to review all places where `getInnstanceFile` is called. If you agree please let me do that if not you can merge.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
> there are a few other places where we don't check for null before to avoid crashing so maybe this is a good opportunity to review all places where `getInstanceFile` is called

You're absolutely right. And there are even more where `formController` methods are called without a `null` check. Last time I looked at it my assessment was that there were so many of those cases that we'd go crazy adding all the `null` checks in and that they would make the code even harder to follow. I think making improvements as we are in that code for other reasons or as crashes are actually experienced by users is a pretty good strategy. It also makes it easier to reason about each one -- sometimes the possible `null`s may be symptoms of something else bad happening. That said, if you think it makes sense to do in a sweeping pass, please do issue a PR.

**Pull request merged**